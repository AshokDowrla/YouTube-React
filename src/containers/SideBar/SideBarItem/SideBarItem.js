import React from 'react'
import { Menu, Icon } from 'semantic-ui-react'
import "./SideBarItem.scss"
import { Link, withRouter } from 'react-router-dom';

class SideBarItem extends React.Component {


    shouldBeHighlighted = () => {
        const { pathname } = this.props.location;
        console.log('pathname = ', pathname);
        if (this.props.path === '/') {
            return pathname === this.props.path;
        }
        return pathname.includes(this.props.path);
    }


    render() {
        //console.log(this.props.path)
        const highlight = this.shouldBeHighlighted() ? 'highlight-item' : null;
        return (
            <Link to={{ pathname: this.props.path }}>
                <Menu.Item className={['sidebar-item', highlight].join(' ')}>

                    <div className='sidebar-item-alignment-container'>

                        <span> <Icon name={this.props.icon} size="large" /> </span>
                        <span>{this.props.label}</span>
                    </div>
                </Menu.Item>
            </Link>
        )

    }
}


export default withRouter(SideBarItem);