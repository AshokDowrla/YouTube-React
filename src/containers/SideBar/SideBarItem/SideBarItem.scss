@import "../../../styles/shared.scss";
.sidebar-item{
    background: #f5f5f5;

    span{
        i.icon{
            color: $grey;
            margin-right: 20px;
        }
    }


    &.highlight-item {
        span {
          font-weight: 600;
        }
    
        i.icon {
          color: $red;
        }
      }
}



.sidebar-item-alignment-container {
    display: flex;
    align-items: center;
  }