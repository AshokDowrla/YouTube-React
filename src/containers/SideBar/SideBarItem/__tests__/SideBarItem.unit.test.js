
import React from 'react';
import { shallow } from 'enzyme';
import SideBarItem from '../SideBarItem';


describe('SideBarItem', () => {


    test('Render non-highlighted SideBarItem', () => {
        const wrapper = shallow(
            <SideBarItem icon='fire' label='Trending' location={location} />
        );
        expect(wrapper).toMatchSnapshot();
    });


    test('Render highlighted SideBarItem', () => {
        const wrapper = shallow(
            <SideBarItem icon='fire'  label='Trending' highlight />
        );
        expect(wrapper).toMatchSnapshot();
    });



});