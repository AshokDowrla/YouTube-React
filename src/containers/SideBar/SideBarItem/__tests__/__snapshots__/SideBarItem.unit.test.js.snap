// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem Render highlighted SideBarItem 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`SideBarItem Render non-highlighted SideBarItem 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
